### DISCUSSION TITLE
HIP-405: Standard scheme for fungible token metadata on HTS

### DISCUSSION THREAD

**Discussion by mgarbs** [hiero-ledger/hiero-improvement-proposals#491] [General]
This discussion is to discuss HIP-405 - Fungible Token Metadata JSON Schema
@paulatcalaxy @Cooper-Kunz 
[master/HIP/hip-405.md](https://github.com/hashgraph/hedera-improvement-proposal/blob/master/HIP/hip-405.md)

**Comment by rocketmay** [comment]
FT metadata is a topic that is becoming increasingly important now that DEX's are live on Hedera.

One thing that is not directly addressed in this HIP that I think is important is discussion around the addition of a dedicated "metadata" field for fungible tokens, to mirror the metadata field on NFTs. Currently projects are using the "memo" field. Continuing the use the memo field as the standard could be done, but it feels a somewhat inappropriate. 

What are people's thoughts on this?

**Reply by robkainz** [reply]
I completely agree. We plan to use memo for our upcoming release, but would rather use metadata.

**Reply by Cooper-Kunz** [reply]
I think that adding a metadata field would be nice, but out of the scope of HIP-405 which is "what should the metadata look like?" as it is 1) already been solicited for feedback for quite some time, and 2) already being used. Generally, I agree that that a metadata field should should be added, but in a separate HIP - which would become the new logical home for storing metadata matching HIP-405's schema. However because HIP-405 is already in use on immutable tokens on mainnet (e.g. [$CLXY](https://hashscan.io/#/mainnet/token/0.0.859814)) then both would need to be supported, like "legacy" NFTs & HIP-412.

**Comment by Cooper-Kunz** [comment]
Hi @mgarbs I see that this was changed from last call to active. Any indication why? There was already a seemingly good discussion here so I also am unsure why this was created - https://github.com/hashgraph/hedera-improvement-proposal/discussions/403

It seems to have been open for an extremely long time without clarity on how to progress forward.

**Reply by mgarbs** [reply]
Hi @Cooper-Kunz last call is a period, usually 2 weeks, where the hip gathers any lingering community input. Because this was an informational hip, it went directly from last call to active. I created this discussion in error if the one you linked was the original discussion. When I was going through hips I found this hip needed a discussion link in its header and didn't have one, which is why I created this discussion.

**Comment by Ashe-Oro** [comment]
Hi All - is there a HIP to add a metadata field to FT class (token_id)? I see it referenced here, but understand this discussion is regarding metadata schema for FTs. @Cooper-Kunz @rocketmay

**Reply by robkainz** [reply]
> Hi All - is there a HIP to add a metadata field to FT class (token_id)? I see it referenced here, but understand this discussion is regarding metadata schema for FTs. @Cooper-Kunz @rocketmay

I have been wondering too about the existence of any other HIP for HTS metadata (both FTs and the NFT "collection" or "class" token - as I call an NFT without a serial number.) In Wallawallet, we list all the tokens (FTs and NFTs together) and use the memo field as described in HIP405 to grab the icon URL for the token images. So even though this HIP says it's not for NFTs, I do use it for the NFT "class" image as it makes sense to do so given Hedera's APIs. Individual NFTs are displayed in the NFT gallery mode that's easily accessible, where HIP412 is used for that metadata. I am not aware of other wallets treating HIP405 yet in the same way we are though, which makes me wonder also if there is another such HIP that's better suited?
